### ISSUE TITLE
PEP 3110 Use new syntax for catching exceptions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8344]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8344|
|Type|enhancement|
|Created|2016-05-22 08:12:27Z|
|Branch|https://github.com/twisted/twisted/tree/8344-rodrigc-pep3110|

PEP 3110 ( https://www.python.org/dev/peps/pep-3110/ ) specifies a new syntax how to catch exceptions which works in Python 2.6 and higher.

The old syntax:
```
except E, N:
```

should be changed to:
```
except E as N:
```

Both syntaxes work in Python 2.6 and Python 2.7
Only the new syntax works in Python 3.x

<details><summary>Searchable metadata</summary>

```
trac-id__8344 8344
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8344_rodrigc_pep3110 8344-rodrigc-pep3110
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1463904747474831 1463904747474831
changetime__1474198049187875 1474198049187875
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/65

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Thanks for the changes.

I have added the required topfile.

I have reverted the xpathparser.py file as the fix should be in the .g file. I think that we should fix this file in a separate ticket, by fixing the .g file and regenerating the .py file

Waiting for the buildbot status to be all green and then will merge.

Thanks again for the cleanup!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set owner to @adiroiban|
|-|-|

I updated the patch with fixes to a few rst files under the docs directory.
Please review: https://github.com/twisted/twisted/pull/69

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Hm... this ticket is for https://github.com/twisted/twisted/pull/65

the branch is ok... just some minor changes before I can land it

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc set owner to @adiroiban|
|-|-|

Yes, I pushed it.  The patch is the latest one.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset cd248d04fea464ad25837155412a47ac3caa6c21
```
#!CommitTicketReference repository="" revision="cd248d04fea464ad25837155412a47ac3caa6c21"
Merge 8344-rodrigc-pep3110: Use new syntax for catching exceptions.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8344
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
